frontend/src/App.vue

```diff
@@ -1,30 +1,96 @@
 <template>
-  <div>
-    <nav class="navbar bg-navbar navbar-dark shadow">
-      <div class="container-fluid">
-        <router-link class="navbar-brand" to="/">SUSE <i>aimaas</i></router-link>
-
+  <nav class="navbar bg-navbar navbar-dark navbar-expand-lg">
+    <div class="container-fluid">
+      <RouterLink class="navbar-brand" to="/">SUSE <i>aimaas</i></RouterLink>
+      <button class="navbar-toggler" type="button" data-bs-toggle="collapse"
+              data-bs-target="#navbarSupportedContent" aria-controls="navbarSupportedContent"
+              aria-expanded="false" aria-label="Toggle navigation">
+        <span class="navbar-toggler-icon"></span>
+      </button>
+      <div class="collapse navbar-collapse" id="navbarSupportedContent">
+        <ul class="navbar-nav me-auto mb-2 mb-lg-0">
+          <SchemaList v-model="activeSchema" :as-dropdown="true" ref="schemalist"></SchemaList>
+        </ul>
       </div>
-    </nav>
-    <div class="container-fluid mt-2">
-      <router-view v-slot="{ Component }">
-        <keep-alive include="EntityList">
-          <component :is="Component"/>
-        </keep-alive>
-      </router-view>
     </div>
+  </nav>
+  <div class="container mt-2">
+    <keep-alive>
+      <router-view></router-view>
+    </keep-alive>
   </div>
 
 </template>
-
+<style>
+.eos-icons {
+  font-size: 120%;
+  line-height: 120%;
+  vertical-align: text-top;
+}
+</style>
 <script>
-import "bootstrap";
+import {computed} from "vue";
+import 'bootstrap/dist/js/bootstrap.min.js';
+import "eos-icons/dist/css/eos-icons.css";
 import "suse-bootstrap5-theme/dist/css/suse.css";
 
+import {api} from "@/api";
+import SchemaList from "@/components/SchemaList";
+
 export default {
   name: 'App',
-  components: {
-
-  }
+  metaInfo: {
+    title: 'SUSE aimaas'
+  },
+  components: {SchemaList,},
+  data: function () {
+    return {
+      api: api,
+      activeSchema: null
+    }
+  },
+  provide() {
+    return {
+      activeSchema: computed(() => this.activeSchema),
+      availableSchemas: computed(() => this.$refs.schemalist.schemas)
+    }
+  },
+  computed: {
+    availableSchemas() {
+      let _avail_schemas = {};
+      if (this.$refs.schemalist.schemas) {
+        for (let schema of this.$refs.schemalist.schemas) {
+          _avail_schemas[schema.slug] = schema;
+        }
+      }
+      return _avail_schemas
+    }
+  },
+  methods: {
+    getSchemaFromApi(schemaSlug) {
+        api.getSchema({slugOrId: schemaSlug}).then(schema => {
+          this.activeSchema = schema;
+        });
+    },
+    getSchemaFromRoute() {
+      let schemaSlug = this.$route.params.schemaSlug || null;
+      if (schemaSlug === undefined || schemaSlug === null) {
+        return null;
+      }
+      try {
+        // First, try to reuse data in storage
```

Comment:
I was  wondering, won't it make sense to put `this.getSchemaFromApi(schemaSlug);` inside of catch?🤔

Reply:
That sounds feasible at the moment. :+1: 
But it will depend on how we decide to handle errors in the API object.